client/uploadable.go

```diff
@@ -95,10 +95,24 @@ type AutoUploadable struct {
 	mutex    sync.Mutex
 }
 
+//ErrorCode for Ditto error response
+type ErrorCode string
+
+//Error code constants
+const (
+	ErrorCodeParameterInvalid ErrorCode = "messages:parameter.invalid"
+	ErrorCodeExecutionFailed            = "messages:execution.failed"
+)
+
 //ErrorResponse is returned from operations handling functions
 type ErrorResponse struct {
-	Status  int
-	Message string
+	Status  int       `json:"status"`
+	Code    ErrorCode `json:"error"`
```

Comment:
Actually, the struct field "Code" is not correct. It should be: ErrorCode or Error for short but that's not a generic code.

Reply:
I will change the field name to "ErrorCode".